app/models/manageiq/providers/cloud_manager.rb

```diff
@@ -109,6 +110,16 @@ def sync_tenants
       end
     end
 
+    def sync_deleted_cloud_tenants
+      return unless source_tenant
+
+      source_tenant.descendants.each do |tenant|
+        next if tenant.source
+
+        tenant.update_attributes(:parent => Tenant.root_tenant)
```

Comment:
I think the deleted tenant should be moved under the tenant based on the provider instead of the root tenant. This would ensure that users of the deleted tenant won't have visibility to objects outside of the provider.